src/Parser/DeathKnight/Frost/Modules/Talents/GatheringStorm.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import calculateEffectiveDamage from 'Parser/Core/calculateEffectiveDamage';
+
+/**
+ *Each Rune spent during Remorseless Winter increases its damage by 15%, and extends its duration by 0.5 sec.
+ */
+
+const DAMAGE_BOOST = .15;
+const DURATION_BOOST = 500;
+const debug = false;
+
+class GatheringStorm extends Analyzer{
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  totalCasts = 0;
+  bonusDamage = 0;
+  totalStacks = 0;
+  currentStacks = 0;
+  extendedDuration = 0
+
+  on_initialized(){
+    this.active = this.combatants.selected.hasTalent(SPELLS.GATHERING_STORM_TALENT.id);
+  }
+
+  on_byPlayer_applybuff(event){
+    const spellId = event.ability.guid;
+    if(spellId !== SPELLS.GATHERING_STORM_TALENT_BUFF.id){
+      return;
+    }
+    this.currentStacks = 1;
+    this.totalCasts++;
+    this.totalStacks++;
+    debug && console.log("applied buff");
+  }
+
+  on_byPlayer_applybuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId !== SPELLS.GATHERING_STORM_TALENT_BUFF.id){
+      return;
+    }
+    this.currentStacks++;
+    this.totalStacks++;
+    debug && console.log(`added buff stack, now at ${this.currentStacks}`);
+  }
+
+  on_byPlayer_damage(event){
+    const spellId = event.ability.guid;
+    if(spellId !== SPELLS.REMORSELESS_WINTER_DAMAGE.id){
+      return;
+    }
+    const boostedDamage = calculateEffectiveDamage(event, (DAMAGE_BOOST * this.currentStacks));
+    this.bonusDamage += boostedDamage;
+    debug && console.log(`boosted damage with ${this.currentStacks} stacks = ${boostedDamage}`);
+  }
+
+  on_byPlayer_removebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId !== SPELLS.GATHERING_STORM_TALENT_BUFF.id){
+      return;
+    }
+    this.currentStacks = 0;
+    debug && console.log(`removed buff`);
+  }
+
+  on_byPlayer_cast(event){
+    const resource = event.classResources[0];
+    if(event.ability.guid === SPELLS.HOWLING_BLAST.id && this.combatants.selected.hasBuff(SPELLS.RIME.id)){
+      this.extendedDuration += DURATION_BOOST;
+      return;
+    }    
+    if(resource.type !== 5){
+      return;
+    }
+    this.extendedDuration = resource.hasOwnProperty("cost") ? this.extendedDuration + (DURATION_BOOST * resource.cost): this.extendedDuration;
```

Comment:
I assume that obliterate would extend it by 1 sec even if it was cast with obliteration since HB with rime also extends it but just in case it doesn't: Resource cost reduction isn't accounted for in `resource.cost` so if this isn't the case you'd have to check for obliteration and manually subtract one rune.

Reply:
Yeah Obliteration counts as two runes spent the same way Rime counts for its cost